distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Write_Extensions.enso

```diff
@@ -37,14 +41,38 @@ polyglot java import org.enso.base.Array_Builder
    Otherwise, the file is created with the encoded text written to it.
 
    The method returns a `File` object for the written file.
+
+   ? Dry Run
+
+     If writing to Output context is not enabled (such as in "Design" mode),
+     then this function will write to a temporary file. This temporary file will
+     be automatically deleted on exit of the Enso process.
+
+     This allows for building the workflow without affecting the real files.
 Text.write : (File|Text) -> Encoding -> Existing_File_Behavior -> Problem_Behavior -> File ! Encoding_Error | Illegal_Argument | File_Error
 Text.write self path encoding=Encoding.utf_8 on_existing_file=Existing_File_Behavior.Backup on_problems=Problem_Behavior.Report_Warning =
     bytes = self.bytes encoding on_problems
-    file = File.new path
-    r = on_existing_file.write file stream->
-        bytes.if_not_error <|
-            stream.write_bytes bytes
-    r.if_not_error file
+
+    actual = File.new path
+
+    is_enabled = Context.Output.is_enabled
+
+    if_existing = if is_enabled then on_existing_file else
```

Comment:
```suggestion
    effective_existing_behaviour = if is_enabled then on_existing_file else
```

`if_existing` is IMO quite a confusing name to me. It sounds like it has Boolean type, but that is not true.